common/djangoapps/util/db.py

```diff
@@ -157,9 +157,11 @@ def __enter__(self):
 
         connection = transaction.get_connection(self.using)
 
-        # TestCase setup nests tests in two atomics. But there shouldn't be any other.
-        # The outermost atomic starts a transaction and so does not have a savepoint, therefore the "- 1".
-        if self.ALLOW_NESTED and (len(connection.savepoint_ids) - (self.atomic_for_testcase_calls - 1)) > 0:
+        # TestCase setup nests tests in two atomics - one for the test class and one for the individual test.
+        # The outermost atomic starts a transaction - so does not have a savepoint.
+        # The inner atomic starts a savepoint around the test.
+        # So, for tests only, there should be exactly one savepoint_id and two atomic_for_testcase_calls.
+        if self.ALLOW_NESTED and (self.atomic_for_testcase_calls - len(connection.savepoint_ids)) > 1:
```

Comment:
You are correct. I forgot to flip the comparator when I changed the operation order. I'll fix.